### PR TITLE
Remove RepositoryRef type

### PR DESCRIPTION
**Pull request by keepsimple1**
Fix issue #67 .  It is like merging `String` and `&str` (if that was possible ;) ). But this one is possible.

### CODE REVIEW

File: radicle-surf/README.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
 # radicle-surf
 
-A code surfing library for VCS file systems 🏄‍♀️🏄‍♂️
+A code surfing library for Git repository 🏄‍♀️🏄‍♂️
```
Comment (FintanH) [inline radicle-surf/README.md:3]:
```suggestion
A code surfing library for Git repositories 🏄‍♀️🏄‍♂️
```

File: radicle-surf/README.md [hunk 2]
```diff
@@ -18,57 +21,8 @@ our [LICENSE](../LICENSE) file.
 
 Join our community disccussions at [radicle.community](https://radicle.community)!
 
+
 # Example
 
 To a taste for the capabilities of `radicle-surf` we provide an example below, but we also
 keep our documentation and doc-tests up to date.
-
-```rust
```
Comment (FintanH) [inline radicle-surf/README.md:26]: Nice :)

File: radicle-surf/src/git/repo.rs [hunk 3]
```diff
@@ -484,57 +462,39 @@ impl<'a> RepositoryRef<'a> {
...
-    pub fn as_ref(&'_ self) -> RepositoryRef<'_> {
-        RepositoryRef { repo_ref: &self.0 }
-    }
-}
-
-impl<'a> From<&'a Repository> for RepositoryRef<'a> {
-    fn from(repo: &'a Repository) -> Self {
-        repo.as_ref()
+    /// Get a reference to the underlying git2 repo.
+    pub(crate) fn git2_repo(&self) -> &git2::Repository {
```
Comment (FintanH) [inline radicle-surf/src/git/repo.rs:490]: I have a thought about this and maybe you can tell me what you think.

Having this public to the crate means that there's a kind of abstraction leak of the repo when used in other modules. So I'm thinking that we could have two `impl Repository` definitions. The first is the public API, it only contains `pub fn`. The second is the crate API, so it only contains `pub(crate) fn` -- or just `fn` for the cases of only needing it in this file.

Then for any uses outside of this module, we add a `pub(crate) fn`, for example, `find_commit` is one that I spotted. That way we limit the access to the internal repository handle, no matter what it is.

File: radicle-surf/src/git/repo.rs [hunk 4]
```diff
@@ -99,36 +99,14 @@ pub enum Error {
...
-///
-/// # Construction
-///
-/// Use the `From<&'a git2::Repository>` implementation to construct a
-/// `RepositoryRef`.
-#[derive(Clone, Copy)]
-pub struct RepositoryRef<'a> {
-    pub(crate) repo_ref: &'a git2::Repository,
+pub struct Repository {
+    repo: git2::Repository,
```
Comment (FintanH) [inline radicle-surf/src/git/repo.rs:103]: nit: I'd rather call this `inner` or `raw`

**Review by FintanH — COMMENTED** [review]
Generally for this change! Thanks for kicking it off :)

Left some thoughts below.

File: radicle-surf/src/git/repo.rs (hunk 3 above)
Comment (keepsimple1) [reply radicle-surf/src/git/repo.rs:490]: Sounds good!

Comment (keepsimple1) [reply radicle-surf/src/git/repo.rs:490]: Do you want me to make this change in this PR or in a separate PR?

Comment (keepsimple1) [reply radicle-surf/src/git/repo.rs:490]: Like we discussed, let's merge this PR first and then do the methods separation in another PR. Thanks.

**Comment by FintanH** [comment]
Squashed the 3 commits and merged https://github.com/radicle-dev/radicle-git/commit/c44348fc3c832c64ec84efc2f75d055ac52a576c

**Comment by keepsimple1** [comment]
Thanks for squashing the commit. Sorry that I forgot to do that earlier.

**Pull request closed without merging**